platforms/hyperledger-indy/scripts/indy-ledger-script/indy-ledger.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+vault_addr=$1
+vault_token=$2
+admin_path=$3
+admin_name=$4
+identity_path=$5
+identity_name=$6
+identity_role=$7
+pool_genesis_path=$8
+
+[ -z "$vault_addr" ] && { echo "Script Failed: Vault Address Missing"; exit 1;} || echo "Vault Address Received";
+[ -z "$vault_token" ] && { echo "Script Failed: Vault Token Missing"; exit 1;} || echo "VAult Token Received";
+[ -z "$admin_path" ] && { echo "Script Failed: Admin DID Path Missing"; exit 1;} || echo "Admin DID Path Received";
+[ -z "$admin_name" ] && { echo "Script Failed: Admin Name Missing"; exit 1;} || echo "Admin Name Received";
+[ -z "$identity_path" ] && { echo "Script Failed: Identity DID Path Missing"; exit 1;} || echo "Identity DID Received";
+[ -z "$identity_name" ] && { echo "Script Failed: Identity DID Name Missing"; exit 1;} || echo "Identity Name Received";
+[ -z "$identity_role" ] && { echo "Script Failed: Identity Role Missing"; exit 1;} || echo "Identity Role Received";
+[ -z "$pool_genesis_path" ] && { echo "Script Failed: Genesis Pool File Path Missing"; exit 1;} || echo "Genesis Pool File Path Received";
+
+export VAULT_ADDR=$vault_addr
+export VAULT_TOKEN=$vault_token
+
+vault login $vault_token
+
+admin_did=$(vault kv get -field=did $admin_path/$admin_name/identity/public/did)
+
+[ -z "$admin_did" ] && { echo "Script Failed"; exit 1;} || echo "Admin DID Acquired";
+admin_seed=$(vault kv get -field=seed $admin_path/$admin_name)
+[ -z "$admin_seed" ] && { echo "Script Failed"; exit 1;} || echo "Admin Seed Acquired";
+identity_did=$(vault kv get -field=did $identity_path/$identity_name/identity/public/did)
+[ -z "$identity_did" ] && { echo "Script Failed"; exit 1;} || echo "DID Acquired";
+identity_seed=$(vault kv get -field=seed $identity_path/$identity_name/identity/private/seed)
+[ -z "$identity_seed" ] && { echo "Script Failed"; exit 1;} || echo "Seed Acquired";
+
+# Creating did files
+echo "{
+\"version\": 1,
+\"dids\": [{
+\"did\": \"$admin_did\", 
+\"seed\": \"$admin_seed\"
+}]
+}" > admindid.txt
+
+echo "{
+\"version\": 1,
+\"dids\": [{
+\"did\": \"$identity_did\", 
+\"seed\": \"$identity_seed\"
+}]
+}" > identitydid.txt
+
+echo "wallet create myIndyWallet key=12345
+wallet open myIndyWallet key=12345
+wallet list
+did import ./admindid.txt
+did import ./identitydid.txt
+did list
+exit" > indy_txn.txt
+
+indy-cli indy_txn.txt > txn_result.txt
+if grep -q 'Following NYM has been received' 'txn_result.txt'
+then
+	echo "Transaction Successful, NYM has been received"
+else
+	echo "Transaction Failed"
+	exit 1
+fi
+
+identity_verkey=$(vault kv get -field=verkey $identity_path/$identity_name/client/public/verif_keys/verification-keys)
+
+
+echo "wallet open myIndyWallet key=12345
+did use $admin_did
+pool create sandboxpool gen_txn_file=./genesis_pool_config
+pool connect sandboxpool
+pool list
+exit" > indy_txn.txt
+
+indy-cli indy_txn.txt > txn_result.txt
+if grep -q 'Following NYM has been received' 'txn_result.txt'
```

Comment:
You just open the pool here, are you sure you get such a message?

Reply:
You told to check only for the success message.
'Following NYM has been received' if this message  is received then the complete script had run successfully.
if not it will display a failure, 
My previous code was more dynamic. it checks for BAtch Execution failed
and it gives the reason whatever error comes on console. rather than just Script failed msg.

Reply:
Do you want me check for the success for each intermediate steps?

Reply:
What I was implying to is that you won't get a 'Following NYM has been received' message from these commands:

wallet open myIndyWallet key=12345
did use $admin_did
pool create sandboxpool gen_txn_file=./genesis_pool_config
pool connect sandboxpool
pool list
exit

Reply:
yes , thats the point.  since the message 'Following NYM has been received' is not received that means the txn are not successful. and the script has failed.
if you want to know the reason, the previous code does that
if grep -q 'Batch execution failed' 'txn_result.txt'
then
	no=$(grep -oP "#\w" 'txn_result.txt' | tail -c 2)
	echo "Script Failed at line: $no"
	echo $(sed "${no}q;d" 'indy_txn.txt')
	echo "REASON: $(sed -n '/Batch execution failed at line/{x;p;d;}; x' txn_result.txt)"
	exit 1
fi
this just greps whatever the reason of failure is and displays it.
because at any failure ,'Batch execution failed' this will be there. 
if all runs fine then at final step it says NYM Transaction received , which is our success.
Correct me if i am wrong, but id this what is required, if not what exactly is needed for the success/failure checks.